### PR TITLE
Bug#1592694 Query plan temp flag set is missed in create_duplicate_we…

### PR DESCRIPTION
**Pull request by robgolebiowski**

### CODE REVIEW

**Comment by laurynas-biveinis** [comment]
Please always update the bug status once you are working on it (assignment, 5.5/5.6/5.7 series, In Progress, Fix Committed, Fix Released, Invalid, Won't Fix etc.). For this bug 5.5 and 5.6 were New, that is, the bug was found on 5.7, and the earlier versions have not been checked yet - that needs to be checked and the earliest fix version determined

File: mysql-test/include/subquery_sj.inc [hunk 1]
```diff
@@ -7142,3 +7142,41 @@ eval EXPLAIN extended $query;
 eval $query;
 
 DROP TABLE t;
+
```
Comment (laurynas-biveinis) [inline mysql-test/include/subquery_sj.inc]: This should go to one of percona slow query log testcases

File: sql/sql_class.cc [hunk 2]
```diff
@@ -4505,6 +4505,7 @@ void THD::inc_status_created_tmp_disk_tables()
 void THD::inc_status_created_tmp_tables()
 {
   status_var.created_tmp_tables++;
+  query_plan_flags|= QPLAN_TMP_TABLE;
```
Comment (laurynas-biveinis) [inline sql/sql_class.cc:4511]: After the fix, this same statement in create_tmp_table becomes redundant and should be removed

**Comment by laurynas-biveinis** [comment]
The commit message should describe the fix

**Comment by laurynas-biveinis** [comment]
Needs a Jenkins run

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

File: sql/sql_class.cc (hunk 2 above)
Comment (robgolebiowski) [reply sql/sql_class.cc:4511]: Thanks a lot. I was going to removed it, but in the end as it turned out I had not.

**Comment by robgolebiowski** [comment]
Jenkin's run: http://jenkins.percona.com/job/mysql-5.7-param/615/

**Comment by robgolebiowski** [comment]
Still needs to check if this problem occurs on 5.5 and 5.6

**Comment by laurynas-biveinis** [comment]
Looks good - please check lower versions and re-submit this patch as a merge

**Comment by robgolebiowski** [comment]
This is merged from https://github.com/percona/percona-server/pull/1332. Self-approving.

**Comment by laurynas-biveinis** [comment]
@robgolebiowski, please update the Launchpad bug

**Pull request merged**